docs/docs/reference/changed/compiler-plugins.md

```diff
@@ -0,0 +1,133 @@
+---
+layout: doc-page
+title: "Changes in Compiler Plugins"
+---
+
+Compiler plugins are supported by Dotty since 0.9. Compared to Scalac, there are
+two notable changes:
+
+- No more support for analyzer plugins
+- Added support for research plugins
+
+[Analyzer plugins][1] in Scalac are executed during type checking to change the
+normal type checking. This is a nice feature for doing research, but for
+production usage, a predictable and consistent type checker is more important.
+
+For experiments and researches that rely on analyzer plugins in Scalac,
+_research plugin_ can be used for the same purpose in Dotty. Research plugins
+are more powerful than Scalac analyzers as it enables plugin authors to
+customize the whole compiler pipeline. That means, you can easily use your
+customized typer to replace the standard typer, or roll your own parser for
+your domain-specific language. Research plugins are only enabled for nightly or
+snaphot releases of Dotty.
+
+The common plugins that add new phases to the compiler pipeline are called
+_standard plugins_ in Dotty. In terms of features, they are similar to
+Scalac plugins, despite minor changes in the API.
+
+## Physical Interface
```

Comment:
Physical interface reads like a USB port, I'd call this section "Using compiler plugins" and have a separate subsection for the command line and for sbt.

Reply:
Maybe we can call it plugin artifact interface?